### PR TITLE
Panel 14

### PR DESCRIPTION
**Pull request by Jiaqi-beep**
NCL Script: https://www.ncl.ucar.edu/Applications/Scripts/panel_14.ncl
NCL Plot: https://www.ncl.ucar.edu/Applications/Images/panel_14_lg.png
Issue: #335 

image

### CODE REVIEW

**Comment by Jiaqi-beep** [comment]
@anissa111 @michaelavs Hi!! I am having trouble with the log scales on the contour plot. Below is the plot when I set the y-axis to log scale, and it doesn't seem like the same scale as the original one (the tick 100000 should be approaching the middle of the y-axis). There are a couple other problems (minor and major tick marks) but I think figuring out the scale to X and Y axis should be the first step. Do you guys have a clue to what the scale should be? The original NCL script doesn't seem to specify this.
image

**Comment by Jiaqi-beep** [comment]
Edited. Y axis is still not exactly right.
image

**Comment by Jiaqi-beep** [comment]
@anissa111 @michaelavs @erinlincoln Hi all! If any of you would have a chance to look at this it would be aweeesome. I fixed the X axis scale and the problem now is with Y axis. I have tried natural logarithmic, log10, and log2, and none is correct. Right now the scale is x**2 but it is still not exactly right. 

Maybe you would have a hunch what the scale is? Let me know!! Thanks!

**Comment by erinlincoln** [comment]
I played with it on Desmos for a bit and I don't think this is exactly right, but if you just are aiming to get close .55x**2.5 goes through all of the points
Screen Shot 2021-06-21 at 1 50 45 PM
.

**Review by erinlincoln — COMMENTED** [review]
I'm not sure if things are displaying strangely on my end, but whenever I plot the graph many of the labels are overlapping. If it isn't showing up for anyone else though, we can ignore it!
Screen Shot 2021-06-28 at 11 06 06 AM

**Comment by michaelavs** [comment]
> I'm not sure if things are displaying strangely on my end, but whenever I plot the graph many of the labels are overlapping. If it isn't showing up for anyone else though, we can ignore it!
> Screen Shot 2021-06-28 at 11 06 06 AM

I believe this is just from your IDE settings. I used to have a similar issue when using Spyder, but once you make html on the plot, it should resolve correctly. If it isn't, then let me know because there may be another issue going on. (for reference, the plot is is displaying correctly in my IDE which is PyCharm)

File: Plots/Panels/NCL_panel_14.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+T2 = T.isel(time=0).sel(lat_t=-30, lon_t=180, method="nearest")
+T3 = T.isel(time=0).sel(lon_t=270, method="nearest")
+T4 = T.isel(time=0).sel(lon_t=200, method="nearest")
+
+##############################################################################
+# Plot:
+fig = plt.figure(figsize=(10, 13))
+
+# Add the subplots
+ax1 = plt.subplot(2, 2, 1)  # upper left cell of grid
```
Comment (michaelavs) [inline Plots/Panels/NCL_panel_14.py:46]: I think using gridspec is a better idea for this subplot due to the extra white space below the plot. 
```suggestion
ax1 = fig.add_subplot(spec[0, 0])  # upper left cell of grid
```

**Review by michaelavs — COMMENTED** [review]
Looks good overall! I would also like to mention increasing the font size on the tick labels for each axes to about 14 from where it is now (12).

File: Plots/Panels/NCL_panel_14.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+# Extract slices of data for each panel
+T1 = T.isel(time=0).sel(lat_t=30, lon_t=180, method="nearest")
+T2 = T.isel(time=0).sel(lat_t=-30, lon_t=180, method="nearest")
+T3 = T.isel(time=0).sel(lon_t=270, method="nearest")
+T4 = T.isel(time=0).sel(lon_t=200, method="nearest")
+
+##############################################################################
+# Plot:
+fig = plt.figure(figsize=(10, 13))
```
[the remark this answers is not in the document]
Comment (michaelavs) [reply Plots/Panels/NCL_panel_14.py:43]:
```suggestion

fig = plt.figure(figsize=(10, 13), constrained_layout=True)
spec = gridspec.GridSpec(ncols=2, nrows=2, figure=fig)
```
For reference with the gridspec comment below

File: Plots/Panels/NCL_panel_14.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+# Make sure subplots are square
+for axes in [ax1, ax2, ax3, ax4]:
+    axes.set_box_aspect(1)
+
+# Plot xy data at upper left and right plots
+ax1.plot(T1, T.z_t, c='black', linewidth=0.5)
+ax2.plot(T2, T.z_t, c='black', linewidth=0.5)
+
+# # Display X axis ticklabels at the top
```
Comment (michaelavs) [inline Plots/Panels/NCL_panel_14.py:62]:
```suggestion
# Display X axis ticklabels at the top
```

**Comment by Jiaqi-beep** [comment]
@michaelavs Thanks for the reviews!

**Comment by anissa111** [comment]
There's still something about the axes here that sits a little weird with me, but I've looked into it again today and cannot find anything. I added a couple comments explaining more about the axes, and when the checks pass I'll approve. If we ever learn more about what's supposed to be going on with axes like this, I suggest we revisit this example in the future.

**Pull request merged**